src/coreclr/jit/hwintrinsiccodegenarm64.cpp

```diff
@@ -765,30 +827,28 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
                                 // into targetReg. Next, do the predicated operation on the targetReg and last,
                                 // use "sel" to select the active lanes based on mask, and set inactive lanes
                                 // to falseReg.
-
                                 assert(targetReg != embMaskOp2Reg);
                                 assert(HWIntrinsicInfo::IsEmbeddedMaskedOperation(intrinEmbMask.id));
 
-                                GetEmitter()->emitIns_R_R(INS_sve_movprfx, EA_SCALABLE, targetReg, embMaskOp1Reg);
-
-                                emitInsHelper(targetReg, maskReg, embMaskOp2Reg);
+                                emitInsMovPrfxHelper(targetReg, maskReg, embMaskOp1Reg, embMaskOp2Reg);
```

Comment:
this was earlier generating unpredicated version, but now is generated predicated one.